### PR TITLE
Implement Node details.

### PR DESCRIPTION
**Pull request by junior1q94**
Closes [35](https://trello.com/c/7SqjCwLw/35-as-a-developer-i-want-to-create-templates-for-the-front-end-so-that-development-of-each-profile-its-easier-to-control-and-adapt)

Implement Node details.

## Testing

Steps to manually verify the change:
1. _Setup_: step to set the environment
2. _Exercise_: How to run the environment and example link: [Test Link](http://127.0.0.1:8000/)
3. _Verify_: You can see the nodes details

### CODE REVIEW

**Review by mariamera — APPROVED** [review]
LGTM

File: paradise_papers_search/fetch_api/models/helpers.py [hunk 1]
```diff
@@ -59,8 +59,10 @@ def fetch_node_details(node_info):
     node_id         = node_info['node_id']
     node            = MODEL_ENTITIES[node_type].nodes.get(node_id=node_id)
     node_details    = node.serialize
+    node_details['node_connections'] = []
```
Comment (oscarpolanco) [inline paradise_papers_search/fetch_api/models/helpers.py]: I think that it work the same without this assignation 🤔 
What you think?

File: paradise_papers_search/static/js/search.js [hunk 2]
```diff
@@ -2,72 +2,128 @@
...
+      $.getJSON(
+        '/fetch/node',
+        {
+          'id': this._node_id,
+          't': this._node_type
+        }
+      )
+      .done(nodeFetch => {
+        this._node_properties = nodeFetch.response.data.node_properties;
+        nodeFetch.response.data.node_connections.forEach(conns => {
```
Comment (oscarpolanco) [inline paradise_papers_search/static/js/search.js:92]: `conns` 🤔 
Is there a more specific name for this?

File: paradise_papers_search/static/js/search.js [hunk 3]
```diff
@@ -240,6 +314,26 @@
...
+
+      // Show node from cache if connections have being fetched
+      if (this._nodesCache.hasOwnProperty(node_id)) {
+        console.log("Node from cache", node);
+        this._currentNode(this._nodesCache[node_id]);
+        return;
+      }
+
+      // Fetch node connections once
+      console.log("Fetching node details", node);
```
Comment (oscarpolanco) [inline paradise_papers_search/static/js/search.js]: Do we need this `console.log`?

File: paradise_papers_search/static/js/search.js [hunk 4]
```diff
@@ -240,6 +314,26 @@
...
 
+    /**
+     * Toggle _currentNode to show on the view
+     */
+    displayNode (node) {
+      let node_id = node._node_id;
+
+      // Show node from cache if connections have being fetched
+      if (this._nodesCache.hasOwnProperty(node_id)) {
+        console.log("Node from cache", node);
```
Comment (oscarpolanco) [inline paradise_papers_search/static/js/search.js]: Can we check this console.log?

File: paradise_papers_search/static/js/search.js [hunk 5]
```diff
@@ -240,6 +314,26 @@
       }
     }
 
+    /**
+     * Toggle _currentNode to show on the view
+     */
+    displayNode (node) {
+      let node_id = node._node_id;
```
Comment (oscarpolanco) [inline paradise_papers_search/static/js/search.js]: If the node_id don't change we can use `const`
What you think?

File: paradise_papers_search/static/js/search.js [hunk 6]
```diff
@@ -269,8 +363,9 @@
         console.log("Fetch error");
       })
     }
+
   }
 
   // Create and bind our SearchApp
   ko.applyBindings(new SearchApp());
-})()
+})();
```
Comment (oscarpolanco) [inline paradise_papers_search/static/js/search.js:376]: 👍

File: paradise_papers_search/templates/index.html [hunk 7]
```diff
@@ -111,42 +109,156 @@ <h1 class="display-3 mb-5">Paradise Papers Search</h1>
...
           </div>
+
         </div>
       </div>
-
     </div> <!-- Search Results Container  -->
 
 
+    <!-- Node Details Modal  -->
+    <div id="nodeDetailsModal" class="modal fade" tabindex="-1" role="dialog" aria-labelledby="nodeDetailsModal" aria-hidden="true">
```
Comment (oscarpolanco) [inline paradise_papers_search/templates/index.html]: What is the purpose of the `aria-lablelleby` here?
Is using the modal own id 🤔

File: paradise_papers_search/fetch_api/models/helpers.py (hunk 1 above)
Comment (junior1q94) [reply paradise_papers_search/fetch_api/models/helpers.py]: I just want to always return a empty array if there is not connections, for convenience when consuming the API

File: paradise_papers_search/static/js/search.js (hunk 2 above)
Comment (junior1q94) [reply paradise_papers_search/static/js/search.js:92]: It could be nodes_connected...

File: paradise_papers_search/templates/index.html (hunk 7 above)
Comment (junior1q94) [reply paradise_papers_search/templates/index.html]: Will amend, it is the label of the modal.

**Review by raulgomezc — APPROVED** [review]

**Review by oscarpolanco — APPROVED** [review]
LGTM

**Pull request merged**